sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitionsSuite.scala

```diff
@@ -42,35 +43,27 @@ class PruneFileSourcePartitionsSuite extends PrunePartitionSuiteBase {
 
   test("PruneFileSourcePartitions should not change the output of LogicalRelation") {
     withTable("test") {
-      withTempDir { dir =>
-        sql(
-          s"""
-            |CREATE EXTERNAL TABLE test(i int)
-            |PARTITIONED BY (p int)
-            |STORED AS parquet
-            |LOCATION '${dir.toURI}'""".stripMargin)
```

Comment:
Do we know why it uses external table before? Is it related to the test coverage here?

Reply:
I explained this in [the other thread](https://github.com/apache/spark/pull/33350/files/f782ce72e149e362d87dba46d7740d0cd3dbf036#r671463471) and I don't think this is related to the test coverage here. Let me know if you think otherwise @viirya @cloud-fan .

Reply:
> Do we still have the test coverage for partition pruning with hive tables?

@cloud-fan you mean `PruneHiveTablePartitionsSuite`? yes it is untouched by this PR.

Reply:
Yea, I saw the comment. It looks making sense and that's also what I read from the test. Just wondering why it uses external table originally.

Reply:
I'm not sure too, IMO the `EXTERNAL` keyword doesn't matter here.